sold.h

```diff
@@ -270,9 +274,23 @@ class Sold {
             ELFBinary* bin = load.bin;
             Elf_Phdr* phdr = load.orig;
             LOG(INFO) << "Emitting code of " << bin->name() << " from " << HexString(ftell(fp)) << " => " << HexString(load.emit.p_offset)
-                      << " + " << HexString(phdr->p_filesz);
+                      << " + " << HexString(phdr->p_filesz) << SOLD_LOG_KEY(delete_unused_DT_STRTAB_);
             EmitPad(fp, load.emit.p_offset);
-            WriteBuf(fp, bin->head() + phdr->p_offset, phdr->p_filesz);
+
+            std::vector<char> buf(phdr->p_filesz, 0);
+            std::memcpy(buf.data(), bin->head() + phdr->p_offset, phdr->p_filesz);
+            if (delete_unused_DT_STRTAB_) {
```

Comment:
This just a comment for future change, not a review request: Ideally, it'd be better if we can discard information of all unused data (e.g., DT_SYMTAB, DT_GNU_HASH, ...). To implement it, it is probably to better to populate list of unnecessary regions first and then fill zeros for these regions.

Reply:
I agree. It sounds good.